cma-cli-enh (#4851)
centreon/centreon-documentation: pp/integrations/plugin-packs/getting-started/how-to-guides/cma/cma-setup.md

2. Update the Centreon Monitoring Agent connector in the following way: in the **Used by command** field, type **Centreon-Monitoring-Agent** and then click **Select all**.

### Create an authentication token

> > Cloud: a token is present by default on your platform and can be used.

Reviewer: ```suggestion

This step only applies to OnPrem platforms. For Centreon Cloud, a default token is provided on the **Administration > Authentication token** page.

```

## Step 2: Prepare the poller

> > Cloud: This step is not necessary if you want to use CMA with the Central poller.

Reviewer: ```suggestion
> Cloud: This step is not necessary if you want to use CMA with the **Central** poller.
```

<Tabs groupId="sync">
<TabItem value="Agent connects to poller" label="Agent connects to poller">

> > Cloud: when using the central poller, the value of **endpoint** will be **engine-centreon-$\{CLOUD_ORG\}.euwest1.centreon.cloud:443**.

Reviewer: ```suggestion
> Cloud: when using the **Central** poller, the value of **endpoint** will be **engine-centreon-$\{CLOUD_ORG\}.euwest1.centreon.cloud:443**.
```

> $\{CLOUD_ORG\} is present in the URL of your Cloud platform: https://$\{CLOUD_ORG\}.euwest1.centreon.cloud/

Reviewer: ```suggestion
> $\{CLOUD_ORG\} is present in the URL of your Cloud platform: https://$\{CLOUD_ORG\}.euwest1.centreon.cloud/
```

<Tabs groupId="sync">
<TabItem value="Agent connects to poller" label="Agent connects to poller">

> > Cloud: when using the central poller, the value of **Poller endpoint** will be **engine-centreon-$\{CLOUD_ORG\}.euwest1.centreon.cloud:443**.

Reviewer: ```suggestion
> Cloud: when using the **Central** poller, the value of **Poller endpoint** will be **engine-centreon-$\{CLOUD_ORG\}.euwest1.centreon.cloud:443**.
```

> $\{CLOUD_ORG\} is present in the URL of your Cloud platform: https://$\{CLOUD_ORG\}.euwest1.centreon.cloud/

Reviewer: ```suggestion
> $\{CLOUD_ORG\} is present in the URL of your Cloud platform: https://$\{CLOUD_ORG\}.euwest1.centreon.cloud/
```

In this mode, there is no interface. As this installer is not a console program, it returns immediately despite not having finished. You have to wait for a message telling you that all is finished.
If you want to have an exit status, you can launch the installer in a powershell session and wait for the exit code. The exit code will be 0 if all is right.

<Tabs groupId="sync">
<TabItem value="before 25.10" label="Before 25.10">

> > Cloud: when using the central poller, the value of **endpoint** will be **engine-centreon-$\{CLOUD_ORG\}.euwest1.centreon.cloud:443**.

Reviewer: ```suggestion
> Cloud: when using the **Central** poller, the value of **endpoint** will be **engine-centreon-$\{CLOUD_ORG\}.euwest1.centreon.cloud:443**.
```

> $\{CLOUD_ORG\} is present in the URL of your Cloud platform: https://$\{CLOUD_ORG\}.euwest1.centreon.cloud/

Reviewer: ```suggestion
> $\{CLOUD_ORG\} is present in the URL of your Cloud platform: https://$\{CLOUD_ORG\}.euwest1.centreon.cloud/
```

To run it in silent mode, you need to set /S as the first argument. You can display a list of arguments with the following command:

```shell
centreon-monitoring-agent.exe /S --help
```

> To escape the **-** character in an argument value, it must be preceded by **--%**.

Reviewer: ```suggestion
To escape the **-** character in an argument value, it must be preceded by **--%**. Example:
```

| --token                    | Authentication token.
If you use the **--install_plugins** option but the download of the plugins fails, the installer will install the plugins embedded in the installer.

</TabItem>
<TabItem value="25.10" label="From 25.10">

> To run it in silent mode, you must specify /VERYSILENT as the first argument.

Reviewer: ```suggestion
To run it in silent mode, you must specify **/VERYSILENT** as the first argument.
```

You can display a list of arguments with the following command line:

```shell
centreon-monitoring-agent.exe /VERYSILENT --help
```
> Cloud: when using the central poller, the value of **endpoint** will be **engine-centreon-$\{CLOUD_ORG\}.euwest1.centreon.cloud:443**.

> $\{CLOUD_ORG\} is present in the URL of your Cloud platform: https://$\{CLOUD_ORG\}.euwest1.centreon.cloud/

Reviewer: ```suggestion
> $\{CLOUD_ORG\} is present in the URL of your Cloud platform: https://$\{CLOUD_ORG\}.euwest1.centreon.cloud/
```

|/LOGLEVEL| "off","critical","error","warning","info","debug","trace". Default: "error"| if /LOGTYPE=file |
|/MAXFILESIZE| Maximum size of the log file before rotation, in MB. Default: 10. If /LOGTYPE=file | |
|/MAXNUMBER| Maximum number of log files. Both of these parameters are required for log rotation to be enabled. Default: 3. If /LOGTYPE=file | |
|/VERSION| Version of centagent.exe |  |                                                                                                                                                                                                                                                      
                                                                                         
If **/PLUGINSRC=auto** and the download fails, the installer will automatically switch to **embedded** mode.

Silent mode execution errors and the output of /VERSION are written to ./installer_output.log.

*Command examples*

Minimal command (required parameters):

```shell
centreon-monitoring-agent-xxx.exe /VERYSILENT /COMPONENTS=agent,plugins /HOST=host_1 /ENDPOINT=localhost:4317 /TOKEN=token_value
```

Command with optional parameters : 
```shell

> centreon-monitoring-agent-xxx.exe /VERYSILENT /REVERSE /COMPONENTS=agent /HOST=agent1 /ENDPOINT=127.0.0.1:4317 /LOGTYPE=File /LOGLEVEL=Debug /LOGFILE="C:\Logs\agent.log" /MAXFILESIZE=20 /MAXNUMBER=5 /ENCRYPTION=true /CERT="C:\certs\agent.crt" /KEY="C:\certs\agent.key" /CA="C:\certs\ca.crt" /COMMONNAME=centreon /TOKEN=token_value

Reviewer 2: Encryption can take no,insecure,full

> centreon-monitoring-agent-xxx.exe /VERYSILENT /REVERSE /COMPONENTS=agent /HOST=agent1 /ENDPOINT=127.0.0.1:4317 /LOGTYPE=File /LOGLEVEL=Debug /LOGFILE="C:\Logs\agent.log" /MAXFILESIZE=20 /MAXNUMBER=5 /ENCRYPTION=true /CERT="C:\certs\agent.crt" /KEY="C:\certs\agent.key" /CA="C:\certs\ca.crt" /COMMONNAME=centreon /TOKEN=token_value

Reviewer 2: ```suggestion
centreon-monitoring-agent-xxx.exe /VERYSILENT /REVERSE /COMPONENTS=agent /HOST=agent1 /ENDPOINT=127.0.0.1:4317 /LOGTYPE=File /LOGLEVEL=Debug /LOGFILE="C:\Logs\agent.log" /MAXFILESIZE=20 /MAXNUMBER=5 /ENCRYPTION=full /CERT="C:\certs\agent.crt" /KEY="C:\certs\agent.key" /CA="C:\certs\ca.crt" /COMMONNAME=centreon /TOKEN=token_value
```

</TabItem>

> </Tabs>
> 
> ### Updating an existing configuration
> 
> <Tabs groupId="sync">
> <TabItem value="Linux" label="Linux">
> Modify the **/etc/centreon-monitoring-agent/centagent.json** file and restart the agent.
> </TabItem>
> <TabItem value="Windows" label="Windows">
> Run **centreon-monitoring-agent-modify.exe** located in the CMA installation directory.
> This can also be done in silent mode
> ```shell
> centreon-monitoring-agent-modify.exe /VERYSILENT
> ```

Reviewer: ```suggestion
<Tabs groupId="sync">
<TabItem value="Linux" label="Linux">

Modify the **/etc/centreon-monitoring-agent/centagent.json** file and restart the agent.

</TabItem>
<TabItem value="Windows" label="Windows">

Run **centreon-monitoring-agent-modify.exe** located in the CMA installation directory.
This can also be done in silent mode
```shell
centreon-monitoring-agent-modify.exe /VERYSILENT
```

```
